components/exercises/chapter-4-oscar/SumOfARange.tsx

```diff
@@ -0,0 +1,167 @@
+  const [range, setRange] = useState(initialRange);
+  /**
+   * Handles the change event of the start select element and updates the range state accordingly.
+   * @param event - The change event of the start select element.
+   */
+
+  function handleChangeStart(event: React.ChangeEvent<HTMLSelectElement>) {
+    setRange((prevState) => {
+      const range = { ...prevState, start: parseInt(event.target.value) };
+      if (range.start <= range.end) {
+        range.step = 1;
+      }
+
+      return range;
+    });
+  }
+
+  /**
+   * Handles the change event of the end select element and updates the range state accordingly.
+   * @param event - The change event of the end select element.
+   */
+  function handleChangeEnd(event: React.ChangeEvent<HTMLSelectElement>) {
+    setRange((prevState) => {
+      const range = {
+        ...prevState,
+        end: parseInt(event.target.value),
+      };
+      if (range.start > range.end) {
+        range.step = -1;
+      }
+      return range;
+    });
+  }
+
+  return (
+    <div>
+      <div className='container grid grid-cols-5 grid-rows-2 gap-3'>
+        {getRangeWithStep(range.start, range.end, range.step)?.map(
+          (num) =>
+            num === animals[num - 1].id && (
+              <figure
+                className='w-24 h-24 flex flex-col text-center justify-center items-center rounded-md space-y-1'
+                key={animals[num - 1].name}
+              >
+                <Image
+                  width={60}
+                  height={60}
+                  src={animals[num - 1].img}
+                  alt={animals[num - 1].name}
+                />
+                <div>{animals[num - 1].name}</div>
+              </figure>
+            )
+        )}
+      </div>
+
+      <div className='flex flex-row'>
+        <div className='flex flex-col flex-[1.5]'>
+          <div className='flex flex-col'>
+            <p className='mt-3 mb-2'>Select an option to display animals</p>
+            <div className='flex flex-row justify-between items-center'>
+              <div className='w-40 relative'>
+                <select
+                  data-testid='Start'
+                  value={range.start}
+                  onChange={handleChangeStart}
+                  className='block appearance-none w-full bg-white border border-gray-300 hover:border-gray-400 px-4 py-2 pr-8 rounded shadow leading-tight focus:outline-none focus:ring focus:border-blue-500'
+                >
+                  <option value='1'>1</option>
+                  <option value='2'>2</option>
+                  <option value='3'>3</option>
+                  <option value='4'>4</option>
+                  <option value='5'>5</option>
+                  <option value='6'>6</option>
+                  <option value='7'>7</option>
+                  <option value='8'>8</option>
+                  <option value='9'>9</option>
+                  <option value='10'>10</option>
```

Comment:
This repetition could be replaced by array.map

Reply:
refactored